qiskit/visualization/circuit_visualization.py

```diff
@@ -333,6 +333,8 @@ def _text_circuit_drawer(
 
     if filename:
         text_drawing.dump(filename, encoding=encoding)
+        if filename.split(".")[1] == "txt":
```

Comment:
I think it would be better if this check is performed before line 335. That would allow us to catch if any filename with incorrect extensions are given. We could then just raise a warning and not save it. What do you think ? 
Also, I think ```==``` in this line should be ```!=```. And maybe you could use ```filename.endswith(".txt")``` instead of split and check ?

Reply:
I should've used `filename.endswith(".txt")`, my bad. About the warning, I was thinking that maybe we should still allow the saving of the file even if the extension is not `.txt` since we are generating a warning and not an error, which is why I kept it after line 335. Do you think it'll be confusing to the user to see a file containing text but without the `.txt` extension? Let me know what you think is best.